course-content/13_working_with_code.qmd

```diff
@@ -0,0 +1,397 @@
+
+
+*Diving deeper*
+
+The first and best way to work though why your code failed is to look at the information provided at the bottom of the output. As in the examples above, much of the time you will have enough information there to give you some ideas about what the cause of the problem is and a starting point for modifying the code and running it again. 
+
+Many times, however, you will find that your code throws errors you still don't understand.  Other times, you would like to get a better understanding of some topic you have come across. For those moments, don't be shy to use Google to get going by typing in the search field those terms that are narrow enough to describe your problem as succinctly as possible.
+
+<center>
+    <img src="images/working-with-code/goog-stackoverflow.png" size=700 style="margin:auto" border=1/>
+    <p style="text-align: center">
+        Global community and solutions to challenges or fixes for bugs
+    </p>
+</center>
+
+Among the more popular entries you'll find on the search engine response page come from stackoverflow or other coding community sites, and from blogs or other sources. It may also pay to look at "videos" among your Google search responses and you will likely see many YouTube offerings of varying degrees of relevance. There is a whole ecosystem of python developers (polished and not-so-much) who offer content that may be of help. These are often very helpful. 
+
+Imagine we were working through the "SyntaxError: EOL while scanning string literal" error above. You know that it's a syntax error, so you examine your code for something that does not look right based on what you know about writing code. What happens if you still are getting an error? Then a Google search might be helpful.
+
+
+<center>
+    <img src="images/working-with-code/google_syntax_error.png" size=500 style="margin:auto" border=1/>
+    <p style="text-align: center">
+        Error deep dive 
+    </p>
+</center>
+
+The highlighted summary information shows us that it understands that we are referring to a python issue, not a general coding one. So that is a good sign. It summarizes from a page on the realpython.com site (https://realpython.com/invalid-syntax-python/). Moreover it suggests different ways that others have extended the search that you might find helpful. While the suggested site itself is not an official python site, it nonetheless is a quality site about python that will likely be worthwhile to review. Once there, you will find more useful information 
```

Comment:
```suggestion
The highlighted summary information shows us that it understands that we are referring to a python issue, not a general coding one. So that is a good sign. It summarizes from a [page on the realpython.com site](https://realpython.com/invalid-syntax-python/). Moreover it suggests different ways that others have extended the search that you might find helpful. While the suggested site itself is not an official python site, it nonetheless is a quality site about python that will likely be worthwhile to review. Once there, you will find more useful information 
```

If you use \[words you want to display\](URL) syntax, you can get the URL linked to the text.